### ISSUE TITLE
Helm chart: istio-sidecar-injector ConfigMap not created if sidecarInjectorWebhook.enabled=false

### ISSUE THREAD

**Issue by ewbankkit** [istio/istio#7806]
**Describe the bug**
<sub><sup>A clear and concise description of what the bug is.</sup></sub>

If `sidecarInjectorWebhook.enabled` is set to `false` when installing the Istio Helm chart (e.g. for Amazon EKS which can't do sidecar injection because of current lack of support for Mutating and Validating Webhook Admission Controllers), the `istio-sidecar-injector` ConfigMap isn't created.
When I come to inject the sidecar manually into my YAML via `istioctl kube-inject` I get
```console
Error: could not find valid configmap "istio-sidecar-injector" from namespace  "istio-system": configmaps "istio-sidecar-injector" not found - Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure istio-inject configmap exists
```

**Expected behavior**
<sub><sup>A clear and concise description of what you expected to happen.</sup></sub>

The `istio-sidecar-injector` ConfigMap should always be created or should be enabled/disabled independently of sidecarInjectorWebhook enabled/disabled.

**Steps to reproduce the bug**
<sub><sup>Steps to reproduce the behavior.</sup></sub>

Run

```console
$ helm install\
  --kubeconfig kubeconfig\
  --name istio\
  --namespace istio-system\
  --values values.yaml\
  install/kubernetes/helm/istio
```

where `values.yaml` defines

```yaml
sidecarInjectorWebhook:
  enabled: false
```

Then run

```console
istioctl kube-inject
```

and you will see the error message.

**Version**
<sub><sup>What version of istio and Kubernetes are you using? Use `istioctl version` and `kubectl version`</sup></sub>

istio 1.0.0
Kubernetes 1.10.x
Helm 2.10.rc.3

**Is Istio Auth enabled or not?**
<sub><sup>Did you install the stable istio.yaml, istio-auth.yaml.... or if using the Helm chart please provide full command line input.</sup></sub>

**Environment**
<sub><sup>Which environment, cloud vendor, OS, etc are you using?</sup></sub>

AWS/Amazon EKS

**Cluster state**
<sub><sup>If you're running on Kubernetes, consider [following the
instructions](http://istio.io/help/bugs/#generating-a-cluster-state-archive)
to generate "istio-dump.tar.gz", then attach it here by dragging and dropping
the file onto this issue.</sup></sub>

**Comment by ewbankkit** [comment]
The `istio-sidecar-injector` ConfigMap was extracted from the `sidecarInjectorWebhook` chart and made "global" in https://github.com/istio/istio/pull/5722.
The creation of the `istio-sidecar-injector` ConfigMap being dependent on `sidecarInjectorWebhook .enabled` was changed via https://github.com/istio/istio/pull/7408 but I can't see anything in the comments of that PR to indicate why that was

**Comment by rafi** [comment]
This has been fixed in #7565
